crypto/src/main/java/io/neow3j/crypto/transaction/RawTransaction.java

```diff
@@ -121,15 +116,41 @@ private void serializeWithoutScripts(BinaryWriter writer) throws IOException {
     @Override
     public void serialize(BinaryWriter writer) throws IOException {
         serializeWithoutScripts(writer);
-        if (this.scripts.size() != 0) {
-            writer.writeSerializableVariable(this.scripts);
-        }
+        writer.writeSerializableVariable(this.scripts);
     }
 
     public abstract void serializeExclusive(BinaryWriter writer) throws IOException;
 
     public abstract void deserializeExclusive(BinaryReader reader) throws IOException, IllegalAccessException, InstantiationException;
 
+    /**
+     * Serializes this transaction to a raw byte array without any scripts. This is the required
```

Comment:
Minor change. It should be written: "This is required if it's the serialized transaction gets signed, e.g., by an external keypair/provider."